dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -135,17 +135,16 @@
 			ambiq,pwrcfg = <&pwrcfg 0x4 0x8>;
 		};
 
-		i2c3: i2c@40053000 {
-			compatible = "ambiq,i2c";
+		spi3: spi@40053000 {
+			compatible = "ambiq,spi";
```

Comment:
compatible needs to be left unset, and node needs to have a generic name like `iom0`, then board can choose which mode to use.

Reply:
I changed that, thanks for suggestion